src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/UWPtoWinAppSDKUpgrade/WinUIBackButtonCodeFixer.cs

```diff
@@ -0,0 +1,109 @@
+            /*
+              {BackButtonMessage}
+            */";
+
+        private const string BackButtonMessage = @$"
+            TODO {WinUIBackButtonAnalyzer.DiagnosticId} Default back button in the title bar does not exist in WinUI3 apps.
+            The tool has generated a custom back button ""{WinUIBackButtonXamlUpdater.NewBackButtonName}"" in the XAML file.
+            Feel free to edit its position, behavior and use the custom back button instead.
+            Read: https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/case-study-1#restoring-back-button-functionality";
+
+        private ILogger<WinUIBackButtonCodeFixer>? _logger;
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public WinUIBackButtonCodeFixer(ILogger<WinUIBackButtonCodeFixer>? logger)
+        {
+            this._logger = logger;
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            if (root is null || !context.Diagnostics.Any())
+            {
+                return;
+            }
+
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+            var lineSpan = diagnostic.Location.GetLineSpan();
+            var declaration = root.FindToken(diagnosticSpan.Start).Parent?.AncestorsAndSelf().OfType<AssignmentExpressionSyntax>().First();
+
+            if (declaration is null)
+            {
+                return;
+            }
+
+            // Register a code action that will invoke the fix.
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    WinUIBackButtonAnalyzer.DiagnosticId,
+                    c => FixBackButton(context.Document, declaration, lineSpan, c),
+                    equivalenceKey: WinUIBackButtonAnalyzer.DiagnosticId),
+                diagnostic);
+        }
+
+        private async Task<Document> FixBackButton(Document document, AssignmentExpressionSyntax backButtonAssignment, FileLinePositionSpan lineSpan, CancellationToken cancellationToken)
+        {
+            _logger?.LogWarning(lineSpan.Path + BackButtonMessage, null);
+            var comment = await CSharpSyntaxTree.ParseText(BackButtonComment, cancellationToken: cancellationToken).GetRootAsync(cancellationToken).ConfigureAwait(false);
+            StatementSyntax newBackButtonAssignmentWithComment;
+
+            if (backButtonAssignment.Left.ToString().Contains("AppViewBackButtonVisibility")
+                && (backButtonAssignment.Right.ToString() == "AppViewBackButtonVisibility.Collapsed"
```

Comment:
Is this checked in the analyzer? If so, you can use the `Properties` member on the diagnostic to put in decisions you've already calculated

Reply:
This is a fix time check and not checked in the analyzer. It is checking whether the code is fixable, otherwise it just posts a comment